src/coinbase/wallet.ts

```diff
@@ -729,4 +673,36 @@ export class Wallet {
     });
     return encryptionKey;
   }
+
+  /**
+   * Returns a WalletAddress object for the given AddressModel.
+   *
+   * @param addressModel - The AddressModel to build the WalletAddress from.
+   * @param index - The index of the AddressModel.
+   * @returns The WalletAddress object.
+   */
+  private buildWalletAddress(addressModel: AddressModel, index: number): WalletAddress {
+    if (!this.master) {
+      return new WalletAddress(addressModel);
+    }
+    const key = this.deriveKey(index);
+    const ethWallet = new ethers.Wallet(convertStringToHex(key.privateKey!));
+    if (ethWallet.address != addressModel.address_id) {
+      throw new InternalError(`Seed does not match wallet`);
+    }
+
+    return new WalletAddress(addressModel, ethWallet);
+  }
+
+  /**
+   * Fetches a Wallet by its ID. The returned wallet can be immediately used for signing operations if backed by a server signer.
+   * If the wallet is not backed by a server signer, the wallet's seed will need to be set before it can be used for signing operations.
+   *
+   * @param wallet_id - The ID of the Wallet to fetch
+   * @returns The fetched Wallet
+   */
+  public static async fetch(wallet_id: string): Promise<Wallet> {
```

Comment:
Is it typical to intermix `static` v.s. instance, and public v.s. private methods?

In my mind we'd have methods ordered by:

```
static
public
private
```

E.g. roughly something that would be aligned with this: https://typescript-eslint.io/rules/member-ordering/

Reply:
good call! update the order, I will also the have eslint change in a follow up PR